gammapy/modeling/models/spatial.py

```diff
@@ -1506,10 +1506,15 @@ def write(self, overwrite=False):
         overwrite: bool, optional
             Overwrite existing file.
             Default is False, which will raise a warning if the template file exists already.
+        filename: str, optional
+            Set the filename of the template  model, if needed. By default, the template model
+            will be saved with the TemplateSpatialModel.filename.
         """
+        if filename is not None:
```

Comment:
In general, what is the expected behaviour. Should the `write` function really modify the attribute? I would think not, but there might be a good reason.
A minima this should be documented.

Reply:
`Models.read` use the filename from the yaml file so write has to change it.